nemo/nemo/collections/nlp/data/language_modeling/megatron/gpt_dataset.py

```diff
@@ -394,14 +402,73 @@ def _get_text(self, idx: int) -> np.ndarray:
                 self.indexed_dataset.get(self.doc_idx[doc_index_l], length=offset_l + self.add_extra_token)
             )
             sample = np.concatenate(sample_list)
-        if len(sample) != (self.seq_length + self.add_extra_token):
```

Comment:
why are you removing this line

Reply:
Because I now check for that condition in line 451:
```python
elif diff < 0: # too short
 #    sample = np.concatenate([sample, np.full((-1 * diff), self.pad_tok_id)])
            logging.info(
                F' > WARNING: Got sample of length: {len(sample)} for sequence length={self.seq_length+self.add_extra_token}, padding the sample to match sequence length'
            )
            sample = np.array(sample, dtype=np.int64)
            sample = np.pad(
                sample, (0, self.seq_length + self.add_extra_token - len(sample)), mode='constant', constant_values=-1
            )
```